src/quote/getQuoteAndBuyCoverInputs.ts

```diff
@@ -0,0 +1,161 @@
+import axios, { AxiosError } from 'axios';
+import dotenv from 'dotenv';
+
+import { calculatePremiumWithCommissionAndSlippage } from '../buyCover/calculatePremiumWithCommissionAndSlippage';
+import {
+  CoverAsset,
+  CoverId,
+  DEFAULT_COMMISSION_RATIO,
+  DEFAULT_SLIPPAGE,
+  MINIMUM_COVER_PERIOD,
+  NEXUS_MUTUAL_DAO_TREASURY_ADDRESS,
+  TARGET_PRICE_DENOMINATOR,
+} from '../constants/buyCover';
+import {
+  Address,
+  CoverRouterQuoteResponse,
+  ErrorApiResponse,
+  GetQuoteApiResponse,
+  GetQuoteResponse,
+  IntString,
+  Integer,
+  PoolCapacity,
+} from '../types';
+
+dotenv.config();
+
+type CoverRouterQuoteParams = {
+  productId: Integer;
+  amount: IntString;
+  period: Integer;
+  coverAsset: CoverAsset;
+};
+
+async function getQuoteAndBuyCoverInputs(
+  productId: Integer,
+  coverAmount: IntString,
+  coverPeriod: Integer, // days
+  coverAsset: CoverAsset,
+  coverBuyerAddress: Address,
+  paymentAsset: CoverAsset = coverAsset,
+  slippage: number = DEFAULT_SLIPPAGE, // 0.1%
+  ipfsData: string = '', // IPFS base32 hash value
+): Promise<GetQuoteApiResponse | ErrorApiResponse> {
+  let result: GetQuoteResponse | undefined = undefined;
+
+  if (!process.env.COVER_ROUTER_URL) {
+    return { result, error: { message: 'Missing COVER_ROUTER_URL env var', data: {} } };
+  }
+
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return { result, error: { message: 'Invalid productId: must be a positive integer', data: {} } };
+  }
+
+  if (!/^\d+$/.test(coverAmount) || parseInt(coverAmount, 10) <= 0) {
+    return { result, error: { message: 'Invalid coverAmount: must be a positive integer string', data: {} } };
+  }
+
+  if (!Number.isInteger(coverPeriod) || coverPeriod < MINIMUM_COVER_PERIOD) {
```

Comment:
🤔 Shouldn't we check for max value as well?

Reply:
oohh good catch